### DISCUSSION TITLE
thread 'tokio-runtime-worker' panicked at 'Expected parent IDs to be set when ordering by parent ID.'

### DISCUSSION THREAD

**Discussion by xandner** [prisma/prisma#16723] [Q&A]
im using prisma in my system.
my system has 3 service. all of 3 services using 1 postgres database.
in one of them database changed and i forgot pull fro database in my other services.
now i receive  """  thread 'tokio-runtime-worker' panicked at 'Expected parent IDs to be set when ordering by parent ID.',  """
i think if reset the database and import backup to database, its solve. but im not sure

**Comment by janpio** [comment]
Your Client does not match your database. You can either undo the change to the database, or you can update all Client to match the new database structure by updating your Prisma Schema and regenerating those Clients.

**Reply by xandner** [reply]
all of my schema is sync and prismaClient regenerated. but  still going error.
this error not raise from all of my rows and just for some rows happen.
if i use find query like:
` select * from reference_products rp where rp.id=${pid} `
in database and run with prisma, its work and everything is ok.
but if i use:
 ```
 await this.prisma.reference_products.findUnique({
                where: {
                    id: pid 
                },
                include: {
                    refrence_product_history: {
                        select: {
                            id: true,
                        },
                    },
                },
            });
```

throw the error:

> Invalid `prisma.reference_products.findUnique()` invocation: Expected parent IDs to be set when ordering by parent ID.
> This is a non-recoverable error which probably happens when the Prisma Query Engine has a panic.
> https://github.com/prisma/prisma/issues/new?body=Hi+Prisma+Team%21+My+Prisma+Client+just+crashed.+ .....
> If you want the Prisma team to look into it, please open the link above 🙏
> To increase the chance of success, please post your schema and a snippet of how you used Prisma Client in the issue.

my schema is:
```
{
  id                          BigInt                        @id @default(autoincrement())
  category                    Int?
  name                        String                        @unique
  content                     String?
  image                       Int?
  refrence_product_history    refrence_product_history[]
}
```
my prisma version: 4.7.1
my posgtgres version: 13
my node version: 14.19.0

**Reply by janpio** [reply]
Did you create all the data via Prisma or did the database exist before?

**Reply by xandner** [reply]
created with prisma

**Reply by janpio** [reply]
Is this reproducible with that query? On all 3 services that you have deployed or only in one of them?
Can you maybe share the schema of these two and also the data that is in it? Then we could maybe reproduce it.

**Reply by pantharshit00** [reply]
Hello @xandner,

Did you get a chance to check @janpio's reply? If you can provide all the requested information here, it would be very helpful for us to reproduce the problem. 

Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Comment by humblecoder** [comment]
FYI, I'm having a similar issue with a Postgres instance.  The bigger problem for me though, is that once this issue happens, it's truly not recoverable:  meaning, I must nuke my DB instance in order to run the app again -- not even a server restart fixes it.  I have no idea what, but it would appear some aspect of the Primsa -> DB communication chain gets corrupted.

I'm certain there are a lot of details I'm missing, but does anyone have a starting point for troubleshooting/resolving that aspect of this issue?

**Reply by jharrell** [reply]
@humblecoder could you open up an issue with as much info as you can gather? That will help our engineering team look into it further.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
